File: system-test/java/src/main/java/com/optivem/shop/testkit/driver/adapter/external/tax/TaxStubDriver.java
package com.optivem.shop.testkit.driver.adapter.external.tax;

import com.optivem.shop.testkit.driver.adapter.external.tax.client.TaxStubClient;
import com.optivem.shop.testkit.driver.adapter.external.tax.client.dtos.ExtGetCountryResponse;
import com.optivem.shop.testkit.driver.port.external.tax.dtos.ReturnsTaxRateRequest;
import com.optivem.shop.testkit.driver.port.shared.dtos.ErrorResponse;
import com.optivem.shop.testkit.common.Converter;
import com.optivem.shop.testkit.common.Result;

public class TaxStubDriver extends BaseTaxDriver<TaxStubClient> {
    public TaxStubDriver(String baseUrl) {
        super(new TaxStubClient(baseUrl));
    }

    @Override
    public void close() {
        client.removeStubs();
        super.close();
    }

    @Override
    public Result<Void, ErrorResponse> returnsTaxRate(ReturnsTaxRateRequest request) {
        var country = request.getCountry();
        var taxRate = Converter.toBigDecimal(request.getTaxRate());

        var response = ExtGetCountryResponse.builder()
                .id(country)
                .taxRate(taxRate)
                .countryName(country)
                .build();

        return client.configureGetCountry(country, response)
                .mapError(ext -> ErrorResponse.builder().message(ext.getMessage()).build());
    }
}

File: system-test/dotnet/Driver.Adapter/External/Tax/TaxStubDriver.cs
using Common;
using Driver.Adapter.External.Tax.Client;
using Driver.Port.External.Tax.Dtos;
using Driver.Port.External.Tax.Dtos.Error;
using Driver.Adapter.External.Tax.Client.Dtos;
using Driver.Adapter.External.Tax.Client.Dtos.Error;

namespace Driver.Adapter.External.Tax;

public class TaxStubDriver : BaseTaxDriver<TaxStubClient>
{
    public TaxStubDriver(string baseUrl) : base(new TaxStubClient(baseUrl))
    {
    }

    public override Task<Result<VoidValue, TaxErrorResponse>> ReturnsTaxRateAsync(ReturnsTaxRateRequest request)
    {
        var country = request.Country!;
        var taxRate = Converter.ToDecimal(request.TaxRate)!.Value;

        var response = new ExtGetCountryResponse
        {
            Id = country,
            TaxRate = taxRate,
            CountryName = country
        };

        return _client.ConfigureGetCountryAsync(response)
            .MapErrorAsync(MapError);
    }
}
